src/structures/ThreadChannel.js

```diff
@@ -0,0 +1,400 @@
+    this.ownerID = data.owner_id;
+
+    /**
+     * How long in minutes after recent activity before the thread is automatically archived
+     * @type {number}
+     */
+    this.autoArchiveDuration = data.thread_metadata.auto_archive_duration;
+
+    /**
+     * The ID of the last message sent in this thread, if one was sent
+     * @type {?Snowflake}
+     */
+    this.lastMessageID = data.last_message_id;
+
+    /**
+     * The timestamp when the last pinned message was pinned, if there was one
+     * @type {?number}
+     */
+    this.lastPinTimestamp = data.last_pin_timestamp ? new Date(data.last_pin_timestamp).getTime() : null;
+
+    /**
+     * The ratelimit per user for this thread in seconds
+     * @type {number}
+     */
+    this.rateLimitPerUser = data.rate_limit_per_user ?? 0;
+
+    /**
+     * The timestamp the thread was last archived or unarchived at
+     * @type {?number}
+     */
+    this.archiveTimestamp = data.thread_metadata.archive_timestamp
+      ? new Date(data.thread_metadata.archive_timestamp).getTime()
+      : null;
+
+    /**
+     * The approximate count of messages in this thread
+     * <info> This value will not count above 50 even when there are more than 50 messages
+     * If you need an approximate value higher than this, use ThreadChannel#messages.cache.size</info>
+     * @type {?number}
+     */
+    this.messageCount = data.message_count;
+
+    /**
+     * The approximate count of users in this thread
+     * <info> This value will not count above 50 even when there are more than 50 members</info>
+     * @type {?number}
+     */
+    this.memberCount = data.member_count;
+
+    if (data.member && this.client.user?.id) this.members.add({ user_id: this.client.user.id, ...data.member });
+    if (data.messages) for (const message of data.messages) this.messages.add(message);
+  }
+
+  /**
+   * A collection of the guild member objects for each of this threads members
+   * @type {Collection<Snowflake, GuildMember>}
+   * @readonly
+   */
+  get guildMembers() {
+    const members = new Collection();
+    this.members.cache.forEach(member => members.set(member.id, member.guildMember));
+    return members;
+  }
+
+  /**
+   * The time the thread was last archived or unarchived at
+   * @type {Date}
+   * @readonly
+   */
+  get archivedAt() {
+    return new Date(this.archiveTimestamp);
+  }
+
+  /**
+   * The parent channel of this thread
+   * @type {?NewsChannel|TextChannel}
+   * @readonly
+   */
+  get parent() {
+    return this.guild.channels.cache.get(this.parentID) ?? null;
+  }
+
+  /**
+   * Adds a member to this thread.
+   * @param {UserResolvable} member The member to add
+   * @param {string} [reason] The reason for adding this member
+   * @returns {Promise<Snowflake>}
+   */
+  addMember(member, reason) {
+    const id = this.client.users.resolveID(member);
+    if (!id) return Promise.reject(new TypeError('INVALID_TYPE', 'member', 'UserResolvable'));
+    return this.client.api
```

Comment:
I took inspiration for this from Guild#addMember where we can't do members#add because that is an existing method on managers. I think the decision there was that members#addMember felt weird?

Reply:
Now resolved with ad8d16c